### PR TITLE
Migration to windows cluster 12812

### PR DESCRIPTION
**Pull request by elink21**
### What does this PR do?
It adds the relevant files to create a new gke cluster running on windows server nodes, scripts written in bash were migrated to powershell and a new Dockerfile was created, also python unit tests were modified to run specific windows tasks.

### Why is this important?
This is required on ticker beam-12812 and it is an important part of the self hosted migration along with linux and mac parts.

### Where should the reviewer start?

You can try running your own cluster with the follow steps:

- Configure your own kubectl client with gke cluster running windows 2019 ltsc nodes.
- Go to .github/dockerized-gh_actions_runners/kubernetes
- Modify repo and token vars inside github-actions-windows-deployment.yml
- Run kubectl apply -f github-actions-windows-deployment.yml 

### How should this be manually tested?
You can also run the docker file from a windows 10 or windows server environment just passing the github token and repo trough and env file, it will create a single self hosted runner, launched under windows OS.

### Any background context you want to provide?
N/A

### What are the relevant tickets?
N/A

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x]  New feature (non-breaking change which adds functionality)
- [ ]  Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [ ]  I have added necessary documentation (if appropriate)
- [x]  I did review existing Pull Requests before submitting mine
- [ ]  I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: .github/dockerized-gh_actions_runners/Kubernetes/github-actions-windows-deployment.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: iis
```
Comment (dannymartinm) [inline .github/dockerized-gh_actions_runners/Kubernetes/github-actions-windows-deployment.yml:4]: Please include more descriptive names/metadata in order to know more about the deployment and pods. Such as: iis-github-actions-runners or something like that. Thanks!

File: .github/dockerized-gh_actions_runners/Kubernetes/github-actions-windows-deployment.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      nodeSelector:
+        kubernetes.io/os: windows
+      containers:
+      - name: iis-server
+        image:  #todo: replace with gcr, for now im using docker hub img
+        env:
+        - name: GITHUB_REPO
+          value: "Modify with your own repo"
+        - name: GITHUB_TOKEN
+          value: "Modify with your token"
```
Comment (dannymartinm) [inline .github/dockerized-gh_actions_runners/Kubernetes/github-actions-windows-deployment.yml:26]: As the token is a sensitive variable, we need to figure out how to use an encrypted variable. Thank you

**Review by dannymartinm — APPROVED** [review]
Awesome work, lgtm!! Thank you very much 🎉

File: .github/dockerized-gh_actions_runners/Kubernetes/github-actions-windows-deployment.yml (hunk 1 above)
Comment (elink21) [reply .github/dockerized-gh_actions_runners/Kubernetes/github-actions-windows-deployment.yml]: Of course, I renamed tags in the last commit, thx.

File: .github/dockerized-gh_actions_runners/Kubernetes/github-actions-windows-deployment.yml (hunk 2 above)
Comment (elink21) [reply .github/dockerized-gh_actions_runners/Kubernetes/github-actions-windows-deployment.yml]: Sure, I implemented something, will talk about it on this evening meeting.

**Review by dannymartinm — APPROVED** [review]
LGTM 🎉  thank you very much!

**Pull request merged**